src/components/Header/Header.tsx

```diff
@@ -1,23 +1,66 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import Link from 'next/link';
 
 import styles from './Header.module.scss';
 
 export const Header = () => {
     const [isMenuOpen, setIsMenuOpen] = useState(false);
+    const menuRef = useRef<HTMLDivElement | null>(null); // Reference for the menu
+    const closeButtonRef = useRef<HTMLButtonElement | null>(null); // Reference for the close button
+    const firstFocusableElementRef = useRef(null); // First focusable element in the menu
```

Comment:
Nitpick: you've added good names for each of these useRef elements, so we can remove the comments beside their definitions. 

It's good programming practice to use comments; often the aim is to try and add reasons for our implementation methods in those comments instead of describing things that we've already written. If you've written a comment that clarifies an obscure or confusing block of code, sometimes it's worth taking a step back and wondering if there's a better way you can write the code so it's understandable from just reading it.

```suggestion
    const menuRef = useRef<HTMLDivElement | null>(null);
    const closeButtonRef = useRef<HTMLButtonElement | null>(null);
    const firstFocusableElementRef = useRef(null);
```